mod/sign/s3.js

```diff
@@ -0,0 +1,149 @@
+/**
+@module /sign/s3
+@requires aws-sdk/client-s3
+@requires aws-sdk/s3-request-presigner
+Signs requests to S3. Provides functions for get, list, delete and put to S3. 
+*/
+
+//The S3 packages are optional. 
+//Need a temporary assignment to determine if they are available.
+let
+  S3Client,
+  PutObjectCommand,
+  GetObjectCommand,
+  DeleteObjectCommand,
+  ListObjectsCommand,
+  getSignedUrl;
+
+//Check for credentials 
+if (!process.env?.AWS_S3_CLIENT) {

```

Comment:
How do we handle public buckets that don't require any credentials?

Reply:
I'll test what happens with public buckets

Reply:
Public buckets can just be hit on the url. [7071f1b](https://github.com/GEOLYTIX/xyz/pull/1698/commits/7071f1b0c6fa9ec563c9c415a5ed5570134cd541) provides a function that returns the public url if no credentials are supplied